frontend/src/metabase/visualizations/lib/LineAreaBarPostRender.js

```diff
@@ -271,6 +271,21 @@ function onRenderValueLabels(chart, formatYValue, [data]) {
     return { x, y, showLabelBelow };
   });
 
+  const formattingSetting = chart.settings["graph.label_value_formatting"];
+  let compact;
+  if (formattingSetting === "compact") {
+    compact = true;
+  } else if (formattingSetting === "full") {
+    compact = false;
+  } else {
+    // for "auto" we use compact if it shortens avg label length by >3 chars
+    const getAvgLength = compact => {
+      const lengths = data.map(d => formatYValue(d.y, { compact }).length);
```

Comment:
Regarding

> For example, since compact formatting always uses "$" rather than "USD", switching currency symbol can cause auto to switch between compact/full formatting. Anyone have clever ideas for how to avoid that?

The only thing i can think of is to pass in `currency_style: "symbol"` here and make a note of the special-casing in both places. Or probably better, extract them into a `const COMPACT_CURRENCY_OPTIONS = { currency_style: "symbol" };` or something and use in both places.

Reply:
great idea! I'll add that